src/Compilers/CSharp/Test/Semantic/Semantics/TopLevelStatementsTests.cs

```diff
@@ -3688,6 +3730,333 @@ void M()
+                // (16,9): error CS0120: An object reference is required for the non-static field, method, or property 'object.ToString()'
+                //         args.ToString(); // 2
+                Diagnostic(ErrorCode.ERR_ObjectRequired, "args.ToString").WithArguments("object.ToString()").WithLocation(16, 9),
+                // (17,9): error CS0119: 'args' is a type, which is not valid in the given context
+                //         args[0].EndsWith(null); // 3
+                Diagnostic(ErrorCode.ERR_BadSKunknown, "args").WithArguments("args", "type").WithLocation(17, 9),
+                // (33,38): error CS0119: 'args' is a type, which is not valid in the given context
+                //             System.Console.WriteLine(args); // 4
+                Diagnostic(ErrorCode.ERR_BadSKunknown, "args").WithArguments("args", "type").WithLocation(33, 38),
+                // (34,13): error CS0120: An object reference is required for the non-static field, method, or property 'object.ToString()'
+                //             args.ToString(); // 5
+                Diagnostic(ErrorCode.ERR_ObjectRequired, "args.ToString").WithArguments("object.ToString()").WithLocation(34, 13),
+                // (35,13): error CS0119: 'args' is a type, which is not valid in the given context
+                //             args[0].EndsWith(null); // 6
+                Diagnostic(ErrorCode.ERR_BadSKunknown, "args").WithArguments("args", "type").WithLocation(35, 13)
+                );
+
+            var testType = ((Compilation)comp).GetTypeByMetadataName("args");
+
+            Assert.False(comp.NullableSemanticAnalysisEnabled); // To make sure we test incremental binding for SemanticModel
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var nameRefs = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(id => id.Identifier.ValueText == "args").ToArray();
+
+            var nameRef = nameRefs[0];
+            Assert.Equal("using alias1 = args;", nameRef.Parent.ToString());
+
+            Assert.Same(testType, model.GetSymbolInfo(nameRef).Symbol);
+            var names = model.LookupNames(nameRef.SpanStart);
+            Assert.Contains("args", names);
+
+            var symbols = model.LookupSymbols(nameRef.SpanStart);
+            Assert.Contains(testType, symbols);
+            Assert.False(symbols.Any(s => s.Kind == SymbolKind.Parameter));
+            Assert.Same(testType, model.LookupSymbols(nameRef.SpanStart, name: "args").Single());
+
+            nameRef = nameRefs[1];
+            Assert.Equal("System.Console.WriteLine(args)", nameRef.Parent.Parent.Parent.ToString());
+            var parameter = model.GetSymbolInfo(nameRef).Symbol;
+            Assert.Equal("System.String[] args", parameter.ToTestDisplayString());
+            Assert.Equal("<simple-program-entry-point>", parameter.ContainingSymbol.ToTestDisplayString());
```

Comment:
>simple-program [](start = 27, length = 14)

Should this name be changed to align with the feature name? Better yet, could we report "$Main()" instead?

Reply:
> Should this name be changed to align with the feature name?

I am planning to make this adjustment separately. It is not primarily related to the goal of this PR. 

---
In reply to: [424087858](https://github.com/dotnet/roslyn/pull/44167#discussion_r424087858) [](ancestors = 424087858)